rest/src/main/groovy/whelk/rest/api/Crud.groovy

```diff
@@ -548,6 +553,17 @@ class Crud extends HttpServlet {
         }
     }
 
+    private void validateJsonLd(Document doc, HttpServletRequest request, HttpServletResponse response) {
+        List<JsonLdValidator.Error> errors = validator.validate(doc.data)
+        if (errors) {
+            def message = errors.collect { it.toStringWithPath() }
+            failedRequests.labels(request.getMethod(), request.getRequestURI(),
+                    HttpServletResponse.SC_BAD_REQUEST.toString()).inc()
+            response.sendError(HttpServletResponse.SC_BAD_REQUEST,
+                    "Invalid JsonLd. Got errors:" + message)
```

Comment:
Maybe return a structured response, i.e. JSON?
(inconsistent with rest of API though)

Reply:
We discussed this as a possible future feature, but skip it for now.